pylint/checkers/unicode.py

```diff
@@ -0,0 +1,713 @@
+                msg = "invalid-unicode-codec"
+            self.add_message(
+                msg,
+                # Currently Nodes will lead to crashes of pylint
+                # node=node,
+                line=codec_definition_line,
+                end_lineno=codec_definition_line,
+                confidence=interfaces.HIGH,
+                col_offset=None,
+                end_col_offset=None,
+            )
+
+        # Check for invalid content (controls/chars)
+        stream.seek(0)
+        for (lineno, line) in enumerate(_fix_utf16_32_line_stream(stream, codec)):
+            if lineno == 0:
+                line = _remove_bom(line, codec)
+
+            # Look for Bidirectional Unicode, if we use unicode
+            if self._is_unicode(codec):
+                for dangerous in BIDI_UNICODE:
+                    if _cached_encode_search(dangerous, codec) in line:
+                        # Note that we don't add a col_offset on purpose:
+                        #   Using these unicode characters it depends on the editor
+                        #   how it displays the location of characters in the line.
+                        #   So we mark the complete line.
+                        self.add_message(
+                            "bidirectional-unicode",
+                            # Currently Nodes will lead to crashes of pylint
+                            # node=node,
+                            line=lineno + 1,
+                            end_lineno=lineno + 1,
+                            # We mark the complete line, as bidi controls make it hard
+                            # to determine the correct cursor position within an editor
+                            col_offset=0,
+                            end_col_offset=_line_length(line, codec),
+                            confidence=interfaces.HIGH,
+                        )
+                        # We look for bidirectional unicode only once per line
+                        # as we mark the complete line anyway
+                        break
+
+            # Look for chars considered bad
+            matches = self._find_line_matches(line, codec)
+            for col, char in matches.items():
+                self.add_message(
+                    char.human_code(),
+                    # Currently Nodes will lead to crashes of pylint
+                    # node=node,
+                    line=lineno + 1,
+                    end_lineno=lineno + 1,
+                    confidence=interfaces.HIGH,
+                    col_offset=col,
+                    end_col_offset=col + len(char.unescaped),
+                )
+
+    def process_module(self, node: nodes.Module) -> None:
+        """Perform the actual check by checking module stream."""
+        with node.stream() as stream:
+            self._process_stream(stream)
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NoneAsciiNamesChecker(BaseChecker):
```

Comment:
Could we create a ``unicode`` module with a file for each checker ?

Reply:
done